src/python/pants/engine/rules.py

```diff
@@ -317,14 +329,26 @@ def _ensure_type_annotation(
   return annotation
 
 
-def rule(*args, cacheable=True) -> Callable:
+PUBLIC_RULE_DECORATOR_ARGUMENTS = set(['name'])
+
+
+def rule_decorator(*args, **kwargs) -> Callable:
   if len(args) != 1 and not inspect.isfunction(args[0]):
     raise ValueError(
       'The @rule decorator expects no arguments and for the function it decorates to be '
       f'type-annotated. Given {args}.'
     )
 
+  if len(set(kwargs) - PUBLIC_RULE_DECORATOR_ARGUMENTS - set(['cacheable'])) != 0:
+    raise UnrecognizedRuleArgument(
+      f"`@rule`s and `@console_rule`s only accept the following keyword arguments: {PUBLIC_RULE_DECORATOR_ARGUMENTS}"
```

Comment:
This message is lying currently since we also accept `cacheable`. Are we lying on purpose? If yes, please add a comment explaining.

Reply:
I'll add a comment explaining this.

Reply:
Thank you :smile: